stubs/txredisapi.pyi

```diff
@@ -34,11 +36,11 @@ class RedisProtocol(protocol.Protocol):
     def get(self, key: str) -> "Deferred[Any]": ...
 
 class SubscriberProtocol(RedisProtocol):
-    def __init__(self, *args, **kwargs): ...
+    def __init__(self, *args: object, **kwargs: object): ...
     password: Optional[str]
-    def subscribe(self, channels: Union[str, List[str]]): ...
-    def connectionMade(self): ...
-    def connectionLost(self, reason): ...
+    def subscribe(self, channels: Union[str, List[str]]) -> "Deferred[None]": ...
+    def connectionMade(self) -> None: ...
+    def connectionLost(self, reason: Failure = ...) -> None: ...
```

Comment:
Where does the default value for `reason` come from?

Reply:
By default it's a failure which holds a `ConnectionDone` instance. If you feel like spelunking:

- https://twistedmatrix.com/documents/current/api/twisted.internet.protocol.Protocol.html#connectionLost
- https://github.com/twisted/twisted/blob/5ed041274f70285299bcb3dcf11da70604416f63/src/twisted/internet/protocol.py#L571
- https://github.com/twisted/twisted/blob/5ed041274f70285299bcb3dcf11da70604416f63/src/twisted/internet/protocol.py#L524
- https://github.com/twisted/twisted/blob/5ed041274f70285299bcb3dcf11da70604416f63/src/twisted/internet/error.py#L221

Reply:
Actually, that is for the abstract `Protocol`. Let me double check that txredisapi doesn't override that somehow...

Reply:
Oh. It does: https://github.com/IlyaSkriblovsky/txredisapi/blob/c89a49829d1c48baa10d03c41d39c261c02ad6e1/txredisapi.py#L335. Moveover it seems like lots of the implementations of `Protocol` in twisted itself don't have a default for `why`.

In that case, let me remove the default argument.

Reply:
https://github.com/matrix-org/synapse/pull/12528/commits/b4198640f0fad24af70a43c982ecfe7ed6dad460